react/week3/todo-app/src/components/GetTodos.js

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from "react";
+import TodoList from "./TodoList";
+import AddTodo from "./AddTodo"
+
+const GetTodos = () => {
+    const API_URL = "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+    const [todoData, setTododata] = useState([])
+    const [isLoading, setIsLoading] = useState(true);
+    useEffect(() => {
+        fetch(API_URL)
+            .then(res => res.json())
+            .then(todoArray => {
+                setIsLoading(prev => prev = false)
+                return setTododata(todoArray)
+            })
+        return () => {
+            setIsLoading((prev) => prev = true);
```

Comment:
This state update will probably throw you an error: "Can't perform an update on an unmounted component".
This is because this state update is run after the component unmounts.
When a component is unmounted and you mount it again in the future, the whole thing is rerendered from the top, which means `isLoading` is reset to `true`.

Here's an article about possible uses for the clean-up method: https://dev.to/otamnitram/react-useeffect-cleanup-how-and-when-to-use-it-2hbm

Reply:
> This state update will probably throw you an error: "Can't perform an update on an unmounted component".
> This is because this state update is run after the component unmounts.
> When a component is unmounted and you mount it again in the future, the whole thing is rerendered from the top, which means `isLoading` is reset to `true`.
> 
> Here's an article about possible uses for the clean-up method: https://dev.to/otamnitram/react-useeffect-cleanup-how-and-when-to-use-it-2hbm

tq for this article...i understood what u meant...i will change it..